apps/webbsite/src/libs/notion/notion.ts

```diff
@@ -0,0 +1,160 @@
+        auth: NOTION_BLOG_INTEGRATION_TOKEN,
+      });
+
+      this.postsDatabaseID = NOTION_BLOG_INTEGRATION_POSTS_DATABASE_ID;
+      this.videosDatabaseID = NOTION_BLOG_INTEGRATION_VIDEOS_DATABASE_ID;
+
+      this.notionAPI = new NotionAPI({
+        authToken: NOTION_TOKEN_V2,
+        activeUser: NOTION_ACTIVE_USER,
+      });
+    } catch (error) {
+      throw new Error('Failed to initialize Notion API client');
+    }
+  }
+
+  async getPosts() {
+    const response = await this.notion.databases.query({
+      database_id: this.postsDatabaseID,
+      filter: {
+        property: 'Published',
+        checkbox: {
+          equals: true,
+        },
+      },
+      sorts: [
+        {
+          property: 'Slug',
+          direction: 'ascending',
+        },
+      ],
+    });
+
+    const posts = await Promise.all(
+      response.results.map(async (post: any) => {
+        if (!post) return;
+
+        const metadata = {
+          id: post.id,
+          title: post.properties.Title.title[0].plain_text,
+          author: await this.notion.users
+            .retrieve({
+              user_id: post.properties.Author.people[0].id,
+            })
+            .then((user: any) => user.name),
+          description: post.properties.Description.rich_text[0].plain_text,
+          published: post.properties.Published.checkbox,
+          slug: post.properties.Slug.formula.string,
+          tags: post.properties.Tags.multi_select.map((tag: any) => tag.name),
+          cover: post.cover.file.url,
+          dateAndTime: {
+            createdDate: formatDate(post.properties.Created.created_time),
+            createdTime: formatTime(post.properties.Created.created_time),
+            lastEditedDate: formatDate(
+              post.properties['Last edited'].last_edited_time
+            ),
+            lastEditedTime: formatTime(
+              post.properties['Last edited'].last_edited_time
+            ),
+          },
+        };
+
+        const recordMap = await this.getPostRecordMap(post.id);
+
+        return {
+          metadata,
+          recordMap,
+        };
+      })
+    );
+
+    return posts;
+  }
+
+  async getVideos() {
+    const response = await this.notion.databases.query({
+      database_id: this.videosDatabaseID,
+      filter: {
+        property: 'Published',
+        checkbox: {
+          equals: true,
+        },
+      },
+    });
+
+    const posts = await Promise.all(
+      response.results.map(async (video: any) => {
+        if (!video) return;
+
+        const metadata = {
+          id: video.id,
+          title: video.properties.Title.title[0].plain_text,
+          published: video.properties.Published.checkbox,
+          link: video.properties.Link.url,
+          tags: video.properties.Tags.multi_select.map((tag: any) => tag.name),
+          cover: video.cover.file.url,
+        };
+
+        return {
+          metadata,
+        };
+      })
+    );
+
+    return posts;
+  }
```

Comment:
Other considerations to improved error handling: 

1. Add error handling for the `notion.databases.query` call. You could add a try-catch block to handle any errors that might occur during the query.

2. Handle errors in the `map` function by using try-catch blocks. If an error occurs during the iteration, you can catch the error and return a default value or log the error.

3. Consider using `Promise.allSettled` instead of `Promise.all` to ensure that all promises are settled, even if some of them fail. This will ensure that the function returns an array of objects for all the videos, regardless of whether there were errors during the iteration.